### ISSUE TITLE
Gene Overlap annotation v/s Most focal gene annotation for CNVs

### ISSUE THREAD

**Issue by kgaonkar6** [PediatricOpenTargets/ticket-tracker#86]
#### What data file(s) does this issue pertain to?
`consensus_seg_annotated_cn_x_and_y.tsv.gz`

#### What release are you using?

v6

#### Put your question or report your issue here.
sex chr cnvs were annotated with status `deep_deletion` when copy.number==0. This might be capturing broader cnv losses instead of gene level. 

For OT  we only annotate all cnv seg files with genes by genomic region overlap ( [default](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/bc06478f3d92a7d256cbf634054669cb86eabcab/analyses/focal-cn-file-preparation/04-prepare-cn-file.R#L30) : remove cnv that don't overlap any gene)  in `04-prepare-cn-file.R`.

We do have a script that annotates the chr arm/cytoband losses `03-add-cytoband-status-consensus.Rmd` and uses the information to find most focal calls in **autosomes** chr in`05-define-most-focal-cn-units.Rmd` but we didn't run the code in v6 release since `run-bedtools.snakemake` is necessary for the chr arm/cytoband level annotation and it takes a long time to run on our larger sample set. 

Should we update the script to also handle sex chr and use just the most focal calls?

**Comment by kgaonkar6** [comment]
Here's my review of the Deep deletion discrepancy in X chr in consensus calls  compared to controlfreec (in pbta_all pedcbio) 

My steps for the analysis are as follows:

1) Filter consensus_cnv for `deep deletions` in `X` chr only and merge with controlfreec and cnvkit annotated files
```r
consensus_cnv_merge <- consensus_cnv %>% 
  # removing ensembl because a gene can have multiple ensembl ids
  select(-ensembl) %>%
  unique() %>%
  left_join(histology,by=c("biospecimen_id"="Kids_First_Biospecimen_ID")) %>%
  # Only keep chrX since we saw a lot of deep deletions in that chr
  filter(grepl("X",cytoband),
         # Only keep deep deletions for review 
         status=="deep deletion") %>% 
  left_join(check_cnvkit) %>%
  left_join(check_freec) 
```

2) Find copy_number for the gene+biospecimen_id (from part1 ) that don't match controlfreec 
```r
deepdel_not_in_freec <-consensus_cnv_merge %>%
  select(biospecimen_id, status,gene_symbol,starts_with("copy_number")) %>% 
  # find calls that don't match
  filter(copy_number != copy_number_freec) 

# Table of copy_number calls in freec and cnvkit
# for gene+biospecimene_id consensus cnv calls of deep deletions 
deepdel_not_in_freec
```

3) Find SV regions overlapping the gene+biospecimen_id (from part2 ) 
to identify how cnvkit+manta (from 2/3 caller) consensus align
```r
### Checking manta calls
```{r}
# select required cols
deepdel_not_in_freec <- deepdel_not_in_freec %>% select(biospecimen_id,gene_symbol,
                                                        status,
                                                        starts_with("copy_number")) %>% unique()

filter_manta <- function(tumor, gene) {
  check_manta <- check_manta %>% filter( Kids.First.Biospecimen.ID.Tumor == tumor,
               grepl(gene,Gene.name)) %>%
    mutate(gene_symbol=gene) %>%
    group_by(gene_symbol) %>%
    summarise(AnnotSV.ID=toString(AnnotSV.ID))
  return(unique(check_manta$`AnnotSV.ID`))
}

# add annotsv.ID
deepdel_not_in_freec[1:100,] %>% 
  mutate(status_manta = map2(biospecimen_id, gene_symbol, ~filter_manta(.x,.y))) %>% 
  unnest() %>% write_tsv("deepdel_consensus_x_chr_100.txt")
```
I did just 100 since SV can overlap a lot of genes we have to grep for `gene` in `Gene.name`
[deepdel_consensus_x_chr_100.txt](https://github.com/PediatricOpenTargets/ticket-tracker/files/6833130/deepdel_consensus_x_chr_100.txt)

### Find SV regions overlapping the gene+biospecimen_id (from part2 ) 
to identify how cnvkit+manta (from 2/3 caller) consensus align with a 50kb filter used in pedcbio.
```r

filter_manta <- function(tumor, gene) {
  check_manta <- check_manta %>% filter( Kids.First.Biospecimen.ID.Tumor == tumor,
               grepl(gene,Gene.name),
               SV.length <= 50000) %>%
    mutate(gene_symbol=gene) %>%
    group_by(gene_symbol) %>%
    summarise(AnnotSV.ID=toString(AnnotSV.ID))
  return(unique(check_manta$`AnnotSV.ID`))
}

# add annotsv.ID
deepdel_not_in_freec[1:100,] %>% 
  mutate(status_manta = map2(biospecimen_id, gene_symbol, ~filter_manta(.x,.y))) %>% 
  unnest() %>% write_tsv("deepdel_consensus_x_chr_100_50kb.txt")
```
It's empty !
[deepdel_consensus_x_chr_100_50kb.txt](https://github.com/PediatricOpenTargets/ticket-tracker/files/6833216/deepdel_consensus_x_chr_100_50kb.txt)

**Comment by jharenza** [comment]
@kgaonkar6 - this can be closed, correct?

**Comment by kgaonkar6** [comment]
Yes, the issue of excessive deep del in chrX was resolved with manta FILTER==PASS in #1114
